Marlin/Marlin_main.cpp

```diff
@@ -5450,6 +5450,83 @@ inline void gcode_M303() {
 
 #endif // PNEUMATICS
 
+  /**
+   * M382: Drop syringe on active or specified tool
+   */
+  inline void gcode_M382() {
+    int8_t current_syringe_pin = -1;
+    uint8_t tool = active_extruder;
+    // Tool number provided
+    if (code_seen('T')) {
+      tool = code_value();
+    }
+    switch(tool) {
+        case 0:
+          OUT_WRITE(SYRINGE0_PIN, HIGH);
+          current_syringe_pin = SYRINGE0_PIN;
+          break;
+        case 1:
+          OUT_WRITE(SYRINGE1_PIN, HIGH);
+          current_syringe_pin = SYRINGE1_PIN;
+          break;
+      // Invalid Tool Number
+      default:
+        SERIAL_ECHO_START;
+        SERIAL_CHAR('T');
+        SERIAL_PROTOCOL_F(tool, DEC);
+        SERIAL_PROTOCOLPGM(" ");
+        SERIAL_ECHOLNPGM(MSG_INVALID_TOOL);
+        break;
+    }
+    // Verbosity Handling
+    if (code_seen('V')) {
+      if (current_syringe_pin != -1) {
+        bool pin_status = digitalRead(current_syringe_pin);
+        SERIAL_PROTOCOLPGM("Syringe ");
+        SERIAL_PROTOCOL_F(tool, DEC);
+        SERIAL_PROTOCOLPGM(" Status: ");
+        SERIAL_PROTOCOLLN(pin_status);
+      }
+    }
+  }
+
+  /**
+   * M383: Retract syringe on active or specified tool
+   */
+  inline void gcode_M383() {
+    uint8_t tool = active_extruder;
+    // Tool number provided
+    if (code_seen('T')) {
+      tool = code_value();
+    }
+    switch(tool) {
+        case 0:
+          OUT_WRITE(SYRINGE0_PIN, LOW);
+          break;
+        case 1:
+          OUT_WRITE(SYRINGE1_PIN, LOW);
+          break;
```

Comment:
I remember now what it was. It can't take a variable as an argument I believe. 👟